test/org/javarosa/core/model/instance/TreeReferenceParentTest.java

```diff
@@ -0,0 +1,52 @@
+package org.javarosa.core.model.instance;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.javarosa.core.model.instance.TestHelpers.buildRef;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TreeReferenceParentTest {
+
+    @Parameterized.Parameter(value = 0)
+    public String testCase;
+
+    @Parameterized.Parameter(value = 1)
+    public String tr;
+
+    @Parameterized.Parameter(value = 2)
+    public String base;
+
+    @Parameterized.Parameter(value = 3)
+    public String expectedRef;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+            {"Parenting absolute refs doesn't change them", "/foo", "/bar", "/foo"},
+            {"Parenting to an empty ref doesn't change them", "../foo", "", "../foo"},
+            {"foo.parent(bar) gives bar/foo", "foo", "bar", "bar/foo"},
```

Comment:
Hm, this kind of kills my hypothesis that parenting with a relative base that has non-parent steps (`foo.parent(../bar)`) is forbidden because the result isn't useful. This has the same shape.

I still think for now a comment in the tests that says we don't quite understand why the limitation exists is enough. Maybe it's vestigial and we can get rid of it at some point. Maybe there is a check against `null` that it's important for somewhere. We don't really need to figure that out now.